openmmtools/multistate/sams.py

```diff
@@ -280,15 +299,19 @@ def _adapt_target_probabilities_validator(instance, scheme):
     log_target_probabilities = _StoredProperty('log_target_probabilities', validate_function=None)
     state_update_scheme = _StoredProperty('state_update_scheme', validate_function=_StoredProperty._state_update_scheme_validator)
     locality = _StoredProperty('locality', validate_function=None)
+    weight_update = _StoredProperty('weight_update', validate_function=None)
     update_stages = _StoredProperty('update_stages', validate_function=_StoredProperty._update_stages_validator)
-    flatness_criteria = _StoredProperty('flatness_criteria', validate_function=_StoredProperty._flatness_criteria_validator)
-    flatness_threshold = _StoredProperty('flatness_threshold', validate_function=None)
+    if self.update_stages == 'two-stage':
+        criteria = _StoredProperty('criteria', validate_function=_StoredProperty._criteria_validator)
     weight_update_method = _StoredProperty('weight_update_method', validate_function=_StoredProperty._weight_update_method_validator)
+    beta_factor = _StoredProperty('beta_factor', validate_function=None)
     adapt_target_probabilities = _StoredProperty('adapt_target_probabilities', validate_function=_StoredProperty._adapt_target_probabilities_validator)
     gamma0 = _StoredProperty('gamma0', validate_function=None)
     logZ_guess = _StoredProperty('logZ_guess', validate_function=None)
 
     def _initialize_stage(self):
+        self.round_trips = 0
+        self._downhill = False
```

Comment:
Could you add a comment about what this `_downhill` variable means?

Reply:
What about :
```
        # The variable _downhill keeps track of which end state the walker has visited more recently.
        # This is used for computing the number of round trips.
        # _downhill = False: the walker is going uphill because it has visited the initial state more recently than the final state
```

Reply:
Sounds great, thanks!